src/rx-query.ts

```diff
@@ -490,6 +486,118 @@ export class RxQueryBase<
         this._limitBufferSize = bufferSize;
         return this;
     }
+
+    enablePersistentQueryCache(backend: QueryCacheBackend) {
+        this._queryCacheBackend = backend;
+        this._persistentQueryCacheLoaded = this._restoreQueryCacheFromPersistedState();
+        return this;
+    }
+
+    private async _restoreQueryCacheFromPersistedState() {
+        if (!this._queryCacheBackend) {
+            // no cache backend provided, do nothing
+            return;
+        }
+
+        if (this._persistentQueryCacheResult) {
+            // we already restored the cache once, no need to run twice
+            return;
+        }
+
+        if (this.mangoQuery.skip) {
+            console.error('The persistent query cache only works on non-skip queries.');
+            return;
+        }
+
+        const persistentQueryId = this.persistentQueryId();
+        const value = await this._queryCacheBackend.getItem<string[] | string>(`qc:${persistentQueryId}`);
+
+        const lwt = (await this._queryCacheBackend.getItem(`qc:${persistentQueryId}:lwt`)) as string ?? RX_META_LWT_MINIMUM;
+
+        const primaryPath = this.collection.schema.primaryPath;
+
+        this._persistentQueryCacheResult = value ?? undefined;
+
+        // if this is a regular query, also load documents into cache
+        if (Array.isArray(value) && value.length > 0) {
+            this._persistentQueryCacheResult = value;
+            this._persistentQueryCacheResultLwt = lwt;
+
+            const persistedQueryCacheIds = new Set(this._persistentQueryCacheResult);
+
+            let docsData: RxDocumentData<RxDocType>[] = [];
+
+            // query all docs updated > last persisted, limit to an arbitrary 1_000_000 (10x of what we consider our largest library)
+            const {documents: changedDocs} = await this.collection.storageInstance.getChangedDocumentsSince(
+              1_000_000,
+              // make sure we remove the monotonic clock (xxx.01, xxx.02) from the lwt timestamp to avoid issues with
+              // lookups in indices (dexie)
```

Comment:
what's a dexie

Reply:
One of the supported storage engines, he uses `dexie` in his tests (and Loki).